controller_manager/controller_manager/spawner.py

```diff
@@ -266,6 +263,38 @@ def main(args=None):
                         )
                         return 1
 
+                if param_file:
```

Comment:
Besides the simplification, why did you move this before `load_controller` method? Is it for the parameter to be ready when controller is created?

Reply:
Yes exactly. It needs to be ready similar to how `type` parameter is set in the L234-262